argus/util.py

```diff
@@ -362,3 +366,51 @@ def get_command(command, command_type=None):
         True: WINDOWS_NANO
     }
 }
+
+
+class skip_on_os(object):
+    """Decorator to skip a method on a specific OS."""
+
+    def __init__(self, target_os_list, reason=""):
+        """Skip the method decorated if we are using the `os_type`.
+
+        :param target_os_listos_type: A list with OS version in on which
+                                      the decorator will take effect
+        """
+        self._target_os_list = target_os_list
+        self._reason = reason
+
+    def __call__(self, function):
+        """Return the `function` decorated."""
+        @six.wraps(function)
+        def wrapper(*args, **kwargs):
+            try:
+                target_self = args[0]
+            except IndexError:
+                raise exceptions.ErrorArgusInvalidDecorator(
+                    "Incompatible use of decorator %s on %s",
+                    "skip_on_os", function.__name__)
+
+            instance_os_type = None
```

Comment:
why initialize with `instance_os_type` since the function will return if used incorrectly?